src/book.cpp

```diff
@@ -144,10 +144,11 @@ void Book::updateFromXml(const pugi::xml_node& node, const std::string& baseDir)
   m_articleCount = strtoull(ATTR("articleCount"), 0, 0);
   m_mediaCount = strtoull(ATTR("mediaCount"), 0, 0);
   m_size = strtoull(ATTR("size"), 0, 0) << 10;
-  Illustration& favicon = getMutableDefaultIllustration();
-  favicon.data = base64_decode(ATTR("favicon"));
-  favicon.mimeType = ATTR("faviconMimeType");
-  favicon.url = ATTR("faviconUrl");
+  const auto favicon = std::make_shared<Illustration>();
```

Comment:
We probably miss a `m_illustrations.clear()` here.
As long as we have zim files with only one illustration/favicon, it is ok.
But if we have several illustratrions we should clear the vector as we will overwrite the first illustration (without any guaranty that the first illustration (in the archive) is the same size than the "default" illustration put in the xml)

Reply:
> We probably miss a `m_illustrations.clear()` here.

No, we don't. [`std::vector::assign()`](https://en.cppreference.com/w/cpp/container/vector/assign) replaces entire contents of the container.

Reply:
Ho, I was thinking `assign` only change one item.
So we are good here.
Do you want to fix the race condition in `Illustration::getData` now or we merge ?

Reply:
I will fix it shortly.